source/Serial/NetworkSerial.cpp

```diff
@@ -75,132 +75,148 @@ NetworkSerial::~NetworkSerial(
+
+    // Reset with respect to dependencies
+    delete( _rx );
+    _rx = nullptr;
+    delete( _tx );
+    _tx = nullptr;
+    delete( _stream_socket );
+    _stream_socket = nullptr;
 }
 
 void
 NetworkSerial::flush(
-	void
-	)
+    void
+    )
 {
-	_current_store_operation = _tx->StoreAsync();
+    _current_store_operation = _tx->StoreAsync();
+    create_task( _current_store_operation )
+        .then( [ this ]( unsigned int value_ )
+    {
+        return _tx->FlushAsync();
+    } )
+        .then( [ this ]( task<bool> task_ )
+    {
+        try
+        {
+            task_.get();
+
+            //detect disconnection
+            if( _current_store_operation->Status == Windows::Foundation::AsyncStatus::Error )
+            {
+                _connection_ready = false;
+                ConnectionLost( L"A fatal error has occurred in NetworkSerial::flush() and your connection has been lost." );
+            }
+        }
+        catch( Platform::Exception ^e )
+        {
+            _connection_ready = false;
+            ConnectionLost( L"A fatal error has occurred in NetworkSerial::flush() and your connection has been lost. Error: " + e->Message );
+        }
+    } );
 }
 
 uint16_t
 NetworkSerial::read(
-	void
-	)
+    void
+    )
 {
-	uint16_t c = static_cast<uint16_t>( -1 );
-
-	if( !connectionReady() ) {
-		return c;
-	}
-
-	if( available() ) {
-		c = _rx->ReadByte();
-	}
-	else if( _current_load_operation->Status != Windows::Foundation::AsyncStatus::Started ) {
-		// Attempt to detect disconnection
-		if( _current_load_operation->Status == Windows::Foundation::AsyncStatus::Error )
-		{
-			_connection_ready = false;
-			ConnectionLost();
-			return -1;
-		}
-
-		_current_load_operation = _rx->LoadAsync( 100 );
-	}
-
-	return c;
+    uint16_t c = static_cast<uint16_t>( -1 );
+
+    if( !connectionReady() ) {
+        return c;
+    }
+
+    if( available() ) {
+        c = _rx->ReadByte();
+    }
+    else if( _current_load_operation->Status != Windows::Foundation::AsyncStatus::Started ) {
+        // Attempt to detect disconnection
+        if( _current_load_operation->Status == Windows::Foundation::AsyncStatus::Error )
+        {
+            _connection_ready = false;
+            ConnectionLost( L"A fatal error has occurred in NetworkSerial::read() and your connection has been lost." );
+            return -1;
+        }
+
+        _current_load_operation = _rx->LoadAsync( 100 );
+    }
+
+    return c;
 }
 
 uint32_t
 NetworkSerial::write(
-	uint8_t c_
-	)
+    uint8_t c_
+    )
 {
-	// Check to see if connection is ready
-	if( !connectionReady() ) { return 0; }
-
-	// Attempt to detect disconnection
-	if( _current_store_operation && _current_store_operation->Status == Windows::Foundation::AsyncStatus::Error )
-	{
-		_connection_ready = false;
-		ConnectionLost();
-		return 0;
-	}
-
-	_tx->WriteByte( c_ );
-	return 1;
+    // Check to see if connection is ready
+    if( !connectionReady() ) { return 0; }
```

Comment:
It could be done, but I have avoided complicating simple classes. I feel that the use of this class implies needing a connection, and a 0 is returned if the data is not written.